components/RPC-Connection.jsx

```diff
@@ -51,10 +51,16 @@ function RPC({ network, path, defaultValue, filter=undefined }) {
 		} else {
 			// Otherwise attempt to connect to RPC
 			const connect = async () => {
-				await syncData(network, path, setReturnValue);
-				// Apply filter to retrieved value if a filter is provided
-				if(filter !== undefined) {
-					applyFilter(defaultValue.toString(), filter, network, setReturnValue)
+				const newValue = await syncData(network, path, setReturnValue);
+				if (newValue === undefined) {
+					 // There was an issue with the request, use default
+					return;
+				} else if (filter !== undefined) {
+					// Apply filter to retrieved value if a filter is provided
+					applyFilter(newValue, filter, network, setReturnValue)
+				} else {
+					// Apply value as-is
+					setReturnValue(newValue);
```

Comment:
This was actually causing an edge case where some 'const' values without filters were always using the default value even though a value was successfully being retrieved from the chain.  I thought the test framework wasn't working at first but it turned out the test was correct.  Unit testing already paying off. :partying_face: